### DISCUSSION TITLE
Drag link form the dia.Element like from the port.

### DISCUSSION THREAD

**Discussion by smokeelow** [clientIO/joint#2408] [Q&A] [answered]
### Introduction

Hi, is it possible to drag link right from the element `dia.Element` ? In the same way it works with the ports? 
Or the only way to handle such case is to use `ui.Halo` ?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
You can use [element tools](https://resources.jointjs.com/tutorial/element-tools).
-`Connect`: https://codepen.io/jointjs/pen/XWzdOQo
-`HoverConnect`: https://codepen.io/jointjs/details/NWMmOVv

**Reply by smokeelow** [reply]
Thanks, exactly what I was looking for
